packages/ag-charts-enterprise/src/series/sunburst/sunburstSeries.ts

```diff
@@ -0,0 +1,502 @@
+                    (tag === TextNodeTag.Primary) === !bottomHalf ? radius : radius - (height - label.height);
+                text.textAlign = 'center';
+                text.textBaseline = bottomHalf ? 'bottom' : 'top';
+                text.translationX = Math.cos(theta) * translationRadius;
+                text.translationY = Math.sin(theta) * translationRadius;
+                text.rotation = bottomHalf ? theta - Math.PI * 0.5 : theta + Math.PI * 0.5;
+            }
+            text.visible = true;
+        };
+
+        this.groupSelection.selectByClass(Text).forEach((text) => {
+            updateText(text.datum, text, text.tag, false);
+        });
+        this.highlightSelection.selectByClass(Text).forEach((text) => {
+            const node: _ModuleSupport.HierarchyNode = text.datum;
+            const isHighlighted = highlightedNode === node;
+            text.visible = isHighlighted;
+            if (text.visible) {
+                updateText(text.datum, text, text.tag, isHighlighted);
+            }
+        });
+    }
+
+    override async update() {
+        await this.updateSelections();
+        await this.updateNodes();
+    }
+
+    private getSectorFormat(node: _ModuleSupport.HierarchyNode, isHighlighted: boolean): AgSunburstSeriesStyle {
+        const { datum, color: fill, depth } = node;
+        const {
+            formatter,
+            ctx: { callbackCache },
+        } = this;
+        if (!formatter || datum == null || depth == null) {
+            return {};
+        }
+
+        const { colorKey, labelKey, sizeKey, stroke, strokeWidth } = this;
+
+        const result = callbackCache.call(formatter, {
+            seriesId: this.id,
+            depth,
+            datum,
+            colorKey,
+            labelKey,
+            sizeKey,
+            fill,
+            stroke,
+            strokeWidth,
+            highlighted: isHighlighted,
+        });
+
+        return result ?? {};
+    }
+
+    override getTooltipHtml(node: _ModuleSupport.HierarchyNode): string {
+        const { tooltip, colorKey, labelKey, sizeKey, id: seriesId } = this;
+        const { datum, depth } = node;
+        if (datum == null || depth == null) {
+            return '';
+        }
+
+        const title = labelKey != null ? datum[labelKey] : undefined;
+
+        const format = this.getSectorFormat(node, false);
+        const color = format?.fill ?? node.color;
+
+        const defaults: AgTooltipRendererResult = {
+            title,
+            backgroundColor: color,
+        };
+
+        if (!tooltip.renderer && !tooltip.format && !title) {
+            return '';
+        }
+
+        return tooltip.toTooltipHtml(defaults, {
+            depth,
+            datum,
+            colorKey,
+            labelKey,
+            sizeKey,
+            title,
+            color,
+            seriesId,
+        });
+    }
+
+    override async createNodeData() {
+        return [];
+    }
```

Comment:
Seeing what is going on in the update processing above, I still think it would be better to break some of that out into this method, and keep the `updateNodes()` and `updateSelections()` as a purer 'apply calculated state to scene-graph'?

When it comes to animation, the separation of calculations vs. application will help I think.